webapp/javascript/pages/ContinuousSingleView.tsx

```diff
@@ -181,6 +188,30 @@ function ContinuousSingleView() {
               annotations={annotations}
               selectionType="single"
               ContextMenu={contextMenu}
+              onHoverDisplayTooltip={(data) => {
+                if (!singleView || !singleView.profile) {
+                  return null;
+                }
+
+                // TODO: when this could be undefined?
+                if (!data.timeLabel) {
+                  return null;
+                }
+
+                const values = prepareTimelineTooltipContent(
+                  singleView.profile,
+                  query,
+                  data
+                );
+
+                if (values.length <= 0) {
+                  return null;
+                }
+
+                return (
+                  <TimelineTooltip timeLabel={data.timeLabel} items={values} />
+                );
+              }}
```

Comment:
can move this function from JSX to the body of Component?

Reply:
done